blog/_posts/2016-02-03-essence-of-scala.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: blog
+post-type: blog
+by: Martin Odersky
+title: The Essence of Scala
+---
+
+What do you get if you boil Scala on a slow flame and wait until all
+incidental features evaporate and only the most concentrated essence
+remains? After doing this for 8 years we believe we have the answer:
+it's DOT, the calculus of dependent object types, that underlies Scala.
+
+A [paper on DOT](http://infoscience.epfl.ch/record/215280) will be
+presented in April at [Wadlerfest](http://events.inf.ed.ac.uk/wf2016),
+an event celebrating Phil Wadler's 60th birthday. There's also a prior
+[technical report](http://arxiv.org/abs/1510.05216v1) by Tiark Rompf
+and Nada Amin describing a slightly different version of the
+calculus. Each version describes a proof of type soundness that has
+been machine-checked for correctness.
+
+## The DOT calculus
+
+A calculus is a kind of mini-language that is small enough to be
+studied formally. Translated to Scala notation, the language covered
+by DOT is described by the following abstract grammar:
+
+    Value       v  =  (x: T) => t            Function
+                      new { x: T => ds }     Object
+
+    Definition  d  =  def a = t              Method definition
+                      type A = T             Type
+
+    Term        t  =  v                      Value
+                      x                      Variable
+                      t1(t2)                 Application
+                      t.a                    Selection
+                      { val x = t1; t2 }     Local definition
+
+    Type        T  =  Any                    Top type
+                      Nothing                Bottom type
+                      x.A                    Selection
+                      (x: T1) => T2          Function
+                      { def a: T }           Method declaration
+                      { type T >: T1 <: T2 } Type declaration
+                      T1 & T2                Intersection
+                      { x => T }             Recursion
+
+The grammar uses several kinds of names:
+
+    x      for (immutable) variables
+    a      for (parameterless) methods
+    A      for types
+
+The full calculus adds to this syntax formal _typing rules_ that assign
+types `T` to terms `t` and formal _evaluation rules_ that describe how a
+program is evaluated. The main difference between the Wadlerfest paper
+and the prior technical report lies in the form of these evaluation
+rules. The technical report uses a "big step" semantics, i.e. it
+defines an abstract of interpreter that explains how terms map to
```

Comment:
"an abstract of interpreter"?
did you mean, "an abstract interpreter"
is "abstract" really needed?